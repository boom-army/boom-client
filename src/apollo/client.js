import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
