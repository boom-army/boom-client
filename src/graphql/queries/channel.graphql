mutation addChannel(
  $mintAuthority: String!
  $name: String!
  $family: String!
  $description: String
  $image: String
  $status: String
  $channelParentId: String
) {
  addChannel(
    mintAuthority: $mintAuthority
    name: $name
    family: $family
    description: $description
    image: $image
    status: $status
    channelParentId: $channelParentId
  ) {
    id
    mintAuthority
    name
    family
    description
    image
    channelParentId
    status
    membersCount {
      count
      avatars
    }
    createdAt
    updatedAt
  }
}

query channels {
  channels {
    id
    mintAuthority
    name
    family
    description
    image
    channelParentId
    status
    verified
    membersCount {
      count
      avatars
    }
  }
}

mutation channelUnlink($channelId: ID!) {
  channelUnlink(channelId: $channelId)
}

fragment TweetData on Tweet {
  id
  text
  tags
  isTweetMine
  commentsCount
  retweetsCount
  isRetweet
  tipsCount
  createdAt
  channel {
    id
  }
  parentTweet {
    id
    text
    user {
      id
      handle
      avatar
      data {
        avatarIsNFT
      }
    }
  }
  files {
    id
    url
  }
  gif {
    id
    title
    fixedHeightUrl
    originalUrl
  }
  nft {
    id
    publicKey
    name
    symbol
    description
    sellerFeeBasisPoints
    externalUrl
    image
    attributes {
      traitType
      value
    }
    collection {
      name
      family
    }
    properties {
      files {
        uri
        type
      }
      category
      creators {
        address
        share
      }
    }
  }
  user {
    id
    publicAddress
    avatar
    handle
    consumerName
    data {
      avatarIsNFT
    }
  }
  reactions {
    id
    emojiId
    skin
    isMine
    count
  }
}

query channelFeed($channelId: ID!, $offset: Int, $limit: Int) {
  channelFeed(channelId: $channelId, offset: $offset, limit: $limit) {
    ...TweetData
  }
}
