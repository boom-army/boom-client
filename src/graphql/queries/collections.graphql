query getCollection($name: String!) {
  getCollection(name: $name)
}
