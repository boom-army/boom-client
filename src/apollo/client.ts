import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
