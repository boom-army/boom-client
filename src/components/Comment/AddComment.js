import React from "react";
import Button from "../../styles/Button";
import UserAvatar from "../UserAvatar";
import useInput from "../../hooks/useInput";
import { ADD_COMMENT } from "../../queries/comment";
import { Box } from "@mui/system";
import { TWEET } from "../../queries/tweet";
import { USER } from "../../queries/client";
import { displayError } from "../../utils";
import { useQuery, useMutation } from "@apollo/client";
import { useSnackbar } from "notistack";
import { NewTweet } from "../Tweet";


const AddComment = ({ id }) => {
  const comment = useInput("");
  const { enqueueSnackbar } = useSnackbar();

  const [addCommentMutation, { loading }] = useMutation(ADD_COMMENT, {
    update: (cache, payload) => {
      const { tweet } = cache.readQuery({
        query: TWEET,
        variables: {
          id,
        },
      });

      let comments = tweet.comments;
      comments = [...comments, payload.data.addComment];

      cache.writeQuery({
        query: TWEET,
        data: {
          tweet: { ...tweet, commentsCount: tweet.commentsCount + 1, comments },
        },
      });
    },
  });

  const handleAddComment = async (e) => {
    e.preventDefault();

    if (!comment.value) return enqueueSnackbar("Write a reply");

    try {
      await addCommentMutation({
        variables: {
          id,
          text: comment.value,
        },
      });

      comment.setValue("");
      return enqueueSnackbar("Your reply has been added", { variant: "success" });
    } catch (err) {
      return displayError(err, enqueueSnackbar);
    }
  };

  const { data } = useQuery(USER);

  return (
    <>
      <NewTweet feed={data?.feed} parentTweet={id} />
    </>
  );
};

export default AddComment;
