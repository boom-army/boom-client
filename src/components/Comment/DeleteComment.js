import React from "react";
import { DELETE_COMMENT } from "../../queries/comment";
import { TWEET } from "../../queries/tweet";
import { TrashIcon } from "../Icons";
import { useMutation } from "@apollo/client";
import { useParams } from "react-router-dom";
import { useSnackbar } from "notistack";

const DeleteComment = ({ id }) => {
  const { tweetId } = useParams();
  const { enqueueSnackbar } = useSnackbar();

  const [deleteCommentMutation, { loading }] = useMutation(DELETE_COMMENT, {
    variables: { id },
    update: (cache, { data: { deleteComment } }) => {
      const { tweet } = cache.readQuery({
        query: TWEET,
        variables: { id: tweetId },
      });

      cache.writeQuery({
        query: TWEET,
        data: {
          tweet: {
            ...tweet,
            commentsCount: tweet.commentsCount - 1,
            comments: tweet.comments.filter(
              (comment) => comment.id !== deleteComment.id
            ),
          },
        },
      });
    },
  });

  const handleDeleteComment = async () => {
    await deleteCommentMutation();
    enqueueSnackbar("Your comment has been deleted", { variant: "success" });
  };

  return <TrashIcon loading={loading} onClick={handleDeleteComment} />;
};

export default DeleteComment;
