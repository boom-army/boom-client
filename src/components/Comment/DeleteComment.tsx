import React from "react";
import { DELETE_COMMENT } from "../../queries/comment";
import { TWEET } from "../../queries/tweet";
import { TrashIcon } from "../Icons";
import { useMutation } from "@apollo/client";
import { useParams } from "react-router-dom";
import { useSnackbar } from "../../contexts/snackbar";
import { Comment, Tweet } from "../../generated/graphql";

interface DeleteProps {
  id: string;
}

const DeleteComment = ({ id }: DeleteProps) => {
  const { tweetId } = useParams();
  const { enqueueSnackbar } = useSnackbar();

  const [deleteCommentMutation, { loading }] = useMutation(DELETE_COMMENT, {
    variables: { id },
    update: (cache, { data: { deleteComment } }) => {
      const { tweet }: any = cache.readQuery({
        query: TWEET,
        variables: { id: tweetId },
      });

      cache.writeQuery({
        query: TWEET,
        data: {
          tweet: {
            ...tweet,
            commentsCount: tweet.commentsCount - 1,
            comments: tweet.comments.filter(
              (comment: Comment) => comment.id !== deleteComment.id
            ),
          },
        },
      });
    },
  });

  const handleDeleteComment = async () => {
    await deleteCommentMutation();
    enqueueSnackbar("Your comment has been deleted", { variant: "success" });
  };

  return <TrashIcon loading={loading} onClick={handleDeleteComment} />;
};

export default DeleteComment;
