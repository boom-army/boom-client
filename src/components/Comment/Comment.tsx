import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import UserAvatar from "../UserAvatar";
import DeleteComment from "./DeleteComment";
import { setDate } from "../../utils";
import { Box } from "@mui/system";
import { styled } from "@mui/material/styles";
// import { Comment } from "../../generated/graphql";

const Wrapper = styled("div")((props) => ({
  display: "flex",
  borderBottom: `1px solid ${props.theme.tertiaryColor}`,
  padding: "1.5rem 1rem 1rem 1rem",

  ".comment-info-user": {
    display: "flex",

    svg: {
      marginLeft: "0.6rem",
      position: "relative",
      top: "3px",
    },
  },

  ".comment-info-user span.username": {
    fontWeight: "500",
  },

  ".comment-info-user span.secondary": {
    paddingLeft: "0.5rem",
    color: `${props.theme.secondaryColor}`,
  },

  "@media screen and (max-width: 430px)": {
    flexDirection: "column",

    ".comment-info-user": {
      fontSize: "0.83rem",
    },

    ".avatar": {
      display: "none",
    },

    ".username": {
      display: "none",
    },

    ".comment-info-user span.secondary": {
      paddingLeft: "0",
      ":first-of-type": {
        paddingRight: "0.6rem",
      },
    },
  },
}));

interface CommentProps {
  id: string;
  text?: string;
  isCommentMine?: boolean;
  user?: any;
  createdAt?: string;
}

const Comment = ({
  id,
  text,
  isCommentMine,
  user,
  createdAt,
}: CommentProps) => {
  // const { id, text, isCommentMine, user, createdAt } = comment;
  const handle = user && user.handle;

  return (
    <Wrapper>
      <Box mr={2}>
        <UserAvatar avatar={user?.avatar} />
      </Box>
      <div className="comment-info">
        <div className="comment-info-user">
          <span className="username">{user && user.consumerName}</span>
          <Link to={`/${handle}`}>
            <span className="secondary">{`@${handle}`}</span>
            <span className="secondary">
              {moment(setDate(createdAt)).fromNow()}
            </span>
          </Link>
          <span>{isCommentMine ? <DeleteComment id={id} /> : null}</span>
        </div>

        <p>{text}</p>
      </div>
    </Wrapper>
  );
};

export default Comment;
