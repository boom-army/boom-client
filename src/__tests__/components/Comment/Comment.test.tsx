import React from "react";
import Comment from "../../../components/Comment/Comment";
import { act, fireEvent, render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { SnackbarProvider } from "notistack";
import { DELETE_COMMENT } from "../../../queries/comment";
import { MockedProvider } from "@apollo/client/testing";

const data: any = {
  id: "cl084xdy815761s0n5omoxoem0",
  text: "testing after changes in input converted to ts",
  tags: [],
  isTweetMine: true,
  commentsCount: 1,
  retweetsCount: 0,
  isRetweet: false,
  tipsCount: "0",
  createdAt: "1646139451616",
  parentTweet: null,
  files: [],
  gif: null,
  nft: null,
  user: {
    id: "ckzny1iv10031lmn568gj3b3n",
    publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
    avatar: "",
    handle: "shy-cloud-4965",
    consumerName: "shy-cloud-4965",
    __typename: "User",
  },
  reactions: [
    {
      id: "cl0c774p371312s0n533u0b29k",
      emojiId: "+1",
      skin: 1,
      isMine: true,
      count: 1,
      __typename: "Reaction",
    },
  ],
  __typename: "Tweet",
};

let user: any = {
  id: "ckzny1iv10031lmn568gj3b3n",
  publicAddress: "JBZ52cKhHiFJdzQBNWnp3Xy2jHgDgkhWwSLWoNgLNxD4",
  avatar: "",
  handle: "shy-cloud-4965",
  consumerName: "shy-cloud-4965",
  __typename: "User",
};

export const __mocks__: any = [
  {
    request: {
      query: DELETE_COMMENT,
      variables: {
        id: "cl084xdy815761s0n5omoxoem0",
      },
    },
    result: {
      data: {},
    },
  },
];

describe("<Comment/> component :", () => {
  test("display <Comment/> component ", async () => {
    Date.now = jest.fn(() => 1482363367071);
    let rendered;
    await act(async () => {
      rendered = render(
        <MockedProvider mocks={__mocks__} addTypename={false}>
          <SnackbarProvider>
            <Comment
              id="cl084xdy815761s0n5omoxoem0"
              text="testing after changes in input converted to ts"
              isCommentMine={true}
              user={user}
              createdAt={Date.now().toString()}
            />
          </SnackbarProvider>
        </MockedProvider>,
        { wrapper: MemoryRouter }
      );
    });
    expect(rendered).toMatchSnapshot();
  });
});
