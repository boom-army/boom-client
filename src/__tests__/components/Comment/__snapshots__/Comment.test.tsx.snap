// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment/> component : display <Comment/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-kvwpwc"
      >
        <div
          class="MuiBox-root css-1ay9vb9"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="comment-info"
        >
          <div
            class="comment-info-user"
          >
            <span
              class="username"
            >
              shy-cloud-4965
            </span>
            <a
              href="/shy-cloud-4965"
            >
              <span
                class="secondary"
              >
                @shy-cloud-4965
              </span>
              <span
                class="secondary"
              >
                22 days ago
              </span>
            </a>
            <span>
              <svg
                class="css-1csj31"
                loading=""
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M20.746 5.236h-3.75V4.25c0-1.24-1.01-2.25-2.25-2.25h-5.5c-1.24 0-2.25 1.01-2.25 2.25v.986h-3.75c-.414 0-.75.336-.75.75s.336.75.75.75h.368l1.583 13.262c.216 1.193 1.31 2.027 2.658 2.027h8.282c1.35 0 2.442-.834 2.664-2.072l1.577-13.217h.368c.414 0 .75-.336.75-.75s-.335-.75-.75-.75zM8.496 4.25c0-.413.337-.75.75-.75h5.5c.413 0 .75.337.75.75v.986h-7V4.25zm8.822 15.48c-.1.55-.664.795-1.18.795H7.854c-.517 0-1.083-.246-1.175-.75L5.126 6.735h13.74L17.32 19.732z"
                  />
                  <path
                    d="M10 17.75c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75zm4 0c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75z"
                  />
                </g>
              </svg>
            </span>
          </div>
          <p>
            testing after changes in input converted to ts
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-kvwpwc"
    >
      <div
        class="MuiBox-root css-1ay9vb9"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-2s90m6-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        class="comment-info"
      >
        <div
          class="comment-info-user"
        >
          <span
            class="username"
          >
            shy-cloud-4965
          </span>
          <a
            href="/shy-cloud-4965"
          >
            <span
              class="secondary"
            >
              @shy-cloud-4965
            </span>
            <span
              class="secondary"
            >
              22 days ago
            </span>
          </a>
          <span>
            <svg
              class="css-1csj31"
              loading=""
              viewBox="0 0 24 24"
            >
              <g>
                <path
                  d="M20.746 5.236h-3.75V4.25c0-1.24-1.01-2.25-2.25-2.25h-5.5c-1.24 0-2.25 1.01-2.25 2.25v.986h-3.75c-.414 0-.75.336-.75.75s.336.75.75.75h.368l1.583 13.262c.216 1.193 1.31 2.027 2.658 2.027h8.282c1.35 0 2.442-.834 2.664-2.072l1.577-13.217h.368c.414 0 .75-.336.75-.75s-.335-.75-.75-.75zM8.496 4.25c0-.413.337-.75.75-.75h5.5c.413 0 .75.337.75.75v.986h-7V4.25zm8.822 15.48c-.1.55-.664.795-1.18.795H7.854c-.517 0-1.083-.246-1.175-.75L5.126 6.735h13.74L17.32 19.732z"
                />
                <path
                  d="M10 17.75c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75zm4 0c.414 0 .75-.336.75-.75v-7c0-.414-.336-.75-.75-.75s-.75.336-.75.75v7c0 .414.336.75.75.75z"
                />
              </g>
            </svg>
          </span>
        </div>
        <p>
          testing after changes in input converted to ts
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
